### PR TITLE
Vectorize flags

### PR DESCRIPTION
**Pull request by KentGrigo**
Issue: https://github.com/anki-geo/ultimate-geography/issues/534
Follow-up on https://github.com/anki-geo/ultimate-geography/pull/577

This PR will vectorize the following flags:

- Belize
- Bolivia
- Costa Rica
- El Salvador
- Peru
- Venezuela

The blurred version of these flags were not vectorized because I wasn't able to blur them properly:

- Bolivia
- El Salvador

This means that the question will be on the rasterized, blurred version of the flag and flip to the vectorized flag.

Note that I've updated many of the flags to correspond to the first flag shown on the countries' Wikipedia pages.

### CODE REVIEW

**Comment by aplaice** [comment]
Thanks for the PR!

Since AFAICT the GitHub diff viewer isn't displaying the changes too conveniently, here are the image changes (old flag and then new):

## Belize :heavy_check_mark: 





The proportions have changed, but this is correct.  (The [proportions _should_ be 3:5](https://en.wikipedia.org/wiki/Flag_of_Belize) as i in the new SVG.)

(AFAICT the source file has undergone a flurry of changes in the past couple of years.)

## Bolivia




The blurred image


This is still the state flag, but with a slightly different coat-of-arms ([old source](https://commons.wikimedia.org/wiki/File:Flag_of_Bolivia_(state).svg) vs. [new source](https://commons.wikimedia.org/wiki/File:Bandera_de_Bolivia_(Estado).svg).).  I'm not quite sure why the state flag linked/displayed on Wikipedia has changed...

I would very strongly prefer the blurred image and the non-blurred one, not to differ (other than blurring), and definitely not significatly, like here.

However, if we were to follow our current guidelines ("first flag") we should switch to the civil flag, so we should resolve #111 first (since the blurring issue might become moot).


## Costa Rica




We're removing the coat of arms (switching from state to civil flag).  To avoid back-and-forth we should resolve #111, first.

## El Salvador :white_check_mark: 




The blurred image:


The blurred and unblurred images now differ slightly (other than in the blur), but this is IMO borderline-OK.

## Peru




We're removing the coat of arms (switching from state to civil flag).  To avoid back-and-forth we should resolve #111, first.

## Venezuela




We're removing the coat of arms (switching from state to civil flag).  To avoid back-and-forth we should resolve #111, first.


<hr/>

Several issues (as mentioned above):

1. IMO the blurred image should be the same (apart from blurring :)) as the non-blurred image.  (Blocking, but potentially superseded by **2** since we might not need blurring for Bolivia.)
2. We should resolve #111 before switching any more flags to avoid annoying (for users) back-and-forth.  (Technically, we're not releasing soon, so we could change and then change back, but it's easy to forget.)
3. In the long run, the blurred images should also be SVGs, but that can wait.

<hr/>

Sorry for being pedantic!

**Comment by axelboc** [comment]
For blurring, I could see us being able to add a [`filter`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feGaussianBlur) tag to the SVG file and apply the filter conditionally with CSS to the paths/texts that need to be blurred. IMO, it'll be easier to do this as part of the vectorising work than to update the blurred PNGs.

**Comment by aplaice** [comment]
> For blurring, I could see us being able to add a [filter](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feGaussianBlur) tag to the SVG file and apply the filter conditionally with CSS to the paths/texts that need to be blurred.

That would be convenient, but I don't think we can apply CSS to SVGs that are `img`s (as opposed to SVGs that are inline). (CSS embedded in the SVG files themselves would work, but not CSS in the main Anki template.)

> IMO, it'll be easier to do this as part of the vectorising work than to update the blurred PNGs.

Yes, we definitely shouldn't update the blurred PNGs, but replace them with something derived from the relevant vector images!

**Comment by axelboc** [comment]
Ah good point, that's a shame 👍

**Comment by ukanuk** [comment]
From CONTRIBUTING.md:

> The name should be blurred using [Inkscape](https://inkscape.org/)'s Gaussian blur effect as explained in https://github.com/anki-geo/ultimate-geography/pull/247.

**Comment by KentGrigo** [comment]
Thanks @ukanuk, but I already checked that out. 
Unfortunately, it only states how to apply the filter, not how to make the selection.
However, I've now had the time to look into it and found a solution.

--------

For future reference, here's a guide on how to blur a selection of an SVG file.

To select the part to be blurred
- Use `Edit paths by nodes (N)`
- Click on the text that should be blurred
  - For Bolivia, I also had to select the shadow of the text

To blur
- Select `Filters > Blurs > Blur...`
- Set `Horizontal blur` and `Vertical blur` to `15`
- Un-check `Blur content only`
- Click `Apply`

**Comment by KentGrigo** [comment]
@aplaice, this should fix 1. and 3.

**Review by aplaice — APPROVED** [review]
Thanks very much! These look great!

I'd still rather we deal with #111 first, but we could, in principle merge and then partially revert.

**Review by axelboc — CHANGES_REQUESTED** [review]
Nice, thanks! So apart from the minor tweak below, I think the only thing left is to switch back to the [state flag of Costa Rica](https://commons.wikimedia.org/wiki/File:Flag_of_Costa_Rica_(state).svg) (but in SVG of course), as per @aplaice's assessment in https://github.com/anki-geo/ultimate-geography/issues/111#issuecomment-1382229132

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -261,7 +261,7 @@ convey as little information as possible to not give away the answers.
 
 ### _Flag_ field
 
-This field must contain a single HTML image element pointing to the SVG or PNG image of a flag - e.g. `<img src="ug-flag-seychelles.svg" />`. The image must be placed in the ` media` folder and named `ug-flag-<country_name>.<svg|png>`. SVG is the preferred format.
+This field must contain a single HTML image element pointing to the SVG image of a flag - e.g. `<img src="ug-flag-seychelles.svg" />`. The image must be placed in the ` media` folder and named `ug-flag-<country_name>.svg`. SVG is the preferred format.
```
[the remark this answers is not in the document]
Comment (KentGrigo) [reply CONTRIBUTING.md]: Nice catch. Thanks.

**Review by aplaice — APPROVED** [review]
AFAICT this looks perfect! LGTM!

**Comment by KentGrigo** [comment]
I tried to re-request a review from both of you (@aplaice & @axelboc), but it keeps remove the other.
I've now change the flag of Costa Rica back to the state version, but now vectorized.
Also, I added a blurred version as the banner says `Republica de Costa Rica`.

**Review by axelboc — APPROVED** [review]
Awesome, thanks a lot!!

**Comment by axelboc** [comment]
Summary of changes for future release notes:

- Vectorize the following flags: Belize, Bolivia (+ blurred version), Costa Rica, El Salvador (+ blurred version), Peru, Venezuela
- Add blurred version of flag of Costa Rica
- Use civil flags instead of state flags for Peru and Venezuela
- Update `CONTRIBUTING` to require flag images to be SVG

**Pull request merged**